### ISSUE TITLE
あ

### ISSUE THREAD

**Issue by AkihikoWatanabe** [AkihikoWatanabe/paper_notes#2406]
https://arxiv.org/abs/2508.06471

**Comment by AkihikoWatanabe** [comment]
元ポスト:https://x.com/grad62304977/status/1954805614011453706?s=46&t=Y6UuIHB0Lv0IpmFAjlc2-Q

**Comment by AkihikoWatanabe** [comment]
- アーキテクチャ
  - MoE / sigmoid gates
    - #1719
    - #1754
  - loss free balanced routing
    - #2442
  - widthを小さく、depthを増やすことでreasoning能力改善
  - GQA w/ partial RoPE
    - #1271
    - #1310
  - Attention Headsの数を2.5倍（何に対して2.5倍なんだ、、？）（96個, 5120次元）にすることで（おそらく）事前学習のlossは改善しなかったがReasoning benchmarkの性能改善
  - QK Normを導入しattentionのlogitsの値域を改善
    - #2443
  - Multi Token Prediction
    - #2444
    - #1620

他モデルとの比較
![image](https://github.com/user-attachments/assets/6085b99e-3a76-432c-a759-91dd4feeb219)

学習部分は後で追記する

**Comment by AkihikoWatanabe** [comment]
- 事前学習データ
  - web
    - 英語と中国語のwebページを利用
    - #1944 と同様にquality scoreyをドキュメントに付与
    - 最も低いquality scoreの文書群を排除し、quality scoreの高い文書群をup sampling
    - 最もquality scoreyが大きい文書群は3.2 epoch分利用
    - 多くのweb pageがテンプレートから自動生成されており高いquality scoreが付与されていたが、MinHashによってdeduplicationできなかったため、 #2445 を用いてdocument embeddingに基づいて類似した文書群を排除
  - Multilingual
    - 独自にクロールしたデータとFineWeb-2 #2109 から多言語の文書群を抽出し、quality classifierを適用することでeducational utilityを定量化し、高いスコアの文書群をupsamplingして利用
  - code
    - githubなどのソースコードhosting platformから収集
    - ソースコードはルールベースのフィルタリングをかけ、その後言語ごとのquality modelsによって、high,middle, lowの3つに品質を分類
    - high qualityなものはupsamplingし、low qualityなものは除外
    - #2446 で提案されているFill in the Middle objectiveをコードの事前学習では適用
    - コードに関連するweb文書も事前学習で収集したテキスト群からルールベースとfasttextによる分類器で抽出し、ソースコードと同様のqualityの分類とサンプリング手法を適用。最終的にフィルタリングされた文書群はre-parseしてフォーマットと内容の品質を向上させた
  - math & science
    - web page, 本, 論文から、reasoning能力を向上させるために、数学と科学に関する文書を収集
  - LLMを用いて文書中のeducational contentの比率に基づいて文書をスコアリングしスコアを予測するsmall-scaleな分類器を学習
  - 最終的に事前学習コーパスの中の閾値以上のスコアを持つ文書をupsampling
- 事前学習は2 stageに分かれており、最初のステージでは、"大部分は"generalな文書で学習する。次のステージでは、ソースコード、数学、科学、コーディング関連の文書をupsamplingして学習する。

上記以上の細かい実装上の情報は記載されていない。

mid-training / post trainingについても後ほど追記する

**Comment by AkihikoWatanabe** [comment]
以下も参照のこと
- #2318
